react/features/base/lastn/middleware.js

```diff
@@ -52,12 +58,27 @@ function _updateLastN({ getState }) {
     const defaultLastN = typeof config.channelLastN === 'undefined' ? -1 : config.channelLastN;
     let lastN = defaultLastN;
 
-    if (audioOnly || appState !== 'active') {
+    if (appState !== 'active') {
         lastN = 0;
+    } else if (audioOnly) {
+        const { screenShares, tileViewEnabled } = state['features/video-layout'];
+        const largeVideoParticipantId = state['features/large-video'].participantId;
+        const largeVideoParticipant
+            = largeVideoParticipantId ? getParticipantById(state, largeVideoParticipantId) : undefined;
+
+        if (!tileViewEnabled && largeVideoParticipant && !largeVideoParticipant.local) {
+            lastN = (screenShares || []).includes(largeVideoParticipantId) ? 1 : 0;
```

Comment:
How do you make sure we don't end up rendering a non-screenshare video?

Reply:
Exactly there. If the participant in large video is not part of the screen sharing set, set last N to 0 (no video) otherwise set to 1. When set to 1, the selected participant (aka, the one in the large view) will be displayed.